h2gis-functions/src/test/java/org/h2gis/functions/io/tsv/TSVDriverTest.java

```diff
@@ -104,7 +104,23 @@ public void testWriteRead() throws SQLException, IOException {
             assertTrue(rs.next());
             assertEquals(20000, rs.getDouble(1), 1e-6);
         }
+    }
 
+    @Test
+    public void testWriteReadGZ() throws SQLException, IOException {
+        Statement stat = connection.createStatement();
+        File tsvFile = new File("target/mytsv_export.gz");
+        stat.execute("DROP TABLE IF EXISTS myTSV");
+        stat.execute("create table myTSV(the_geom GEOMETRY, idarea int primary key)");
+        stat.execute("insert into myTSV values('POLYGON ((-10 109, 90 109, 90 9, -10 9, -10 109))', 1)");
+        stat.execute("insert into myTSV values('POLYGON ((90 109, 190 109, 190 9, 90 9, 90 109))', 2)");
+        stat.execute("CALL TSVWrite('target/mytsv_export.gz', 'myTSV')");
+        assertTrue(tsvFile.exists());
+        stat.execute("CALL TSVRead('target/mytsv_export.gz', 'TSV_IMPORT', true);");
+        try (ResultSet rs = stat.executeQuery("select SUM(ST_AREA(the_geom::GEOMETRY)) from TSV_IMPORT")) {
```

Comment:
With this try ressource, if the `stat.executeQuery` fails with an exception, failing won't be detected